### PR TITLE
Update getting started http guide

### PR DESCRIPTION
**Pull request by mario-guerra**
Revised and expanded version of the pet store HTTP example.

### CODE REVIEW

**Comment by mario-guerra** [comment]
@markcowl what do I need to do to properly install and run prettier? 

I tried following [these instructions](https://github.com/microsoft/typespec/blob/8f837b11a3cfa7c95b1658c87a76c536cc483279/packages/prettier-plugin-typespec/README.md) but I just get an error:

```code
.\node_modules\.bin\prettier --write .
[error] Cannot find module 'C:\Users\marioguerra\Work\microsoft-typespec\packages\prettier-plugin-typespec\dist\index.js' imported from C:\Users\marioguerra\Work\microsoft-typespec\node_modules\prettier\index.mjs
```

**Review by timotheeguerin — CHANGES_REQUESTED** [review]
I think things needs to be restructured and the sidebar updated. The guides pages don't show up in the menu

**Comment by mario-guerra** [comment]
@timotheeguerin where can I review how the changes would show up on the website? The preview link is showing me the current version of the tutorial.

**Comment by timotheeguerin** [comment]
Make sure to select `next` version 
image

**Comment by mario-guerra** [comment]
Thanks I see it now. This is how I intended it to flow, I thought having each step show up in the sidebar would be confusing. But we can try it and see what we think. 

Aside from that, what restructuring do you think it needs?

**Comment by timotheeguerin** [comment]
Looking at the first page and the fact that docusausus already has prev/next links below i would most likely never see the `Next: setup` link and dig in the UI 
image

File: docs/getting-started/getting-started-http-03-versioning.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
```
Comment (timotheeguerin) [inline docs/getting-started/getting-started-http-03-versioning.md:1]: Feel like versioning should be the last thing in the guide

File: docs/getting-started/getting-started-http-04-resources-routes.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+model Error {
+  message: string;
+}
+```
+
+- **Error**: This model defines the structure of an error response, with a `message` property.
+
+#### Step 4: Define the ToyOperations Interface
+
+```typescript
```
Comment (timotheeguerin) [inline docs/getting-started/getting-started-http-04-resources-routes.md:158]:
````suggestion
```tsp
````

all of those should be marked as `tsp` or `typespec` to get correct syntax highlighting

File: docs/getting-started/getting-started-http-06-headers.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  op read(@path petId: int32, @header ifMatch?: string): {
+    @statusCode statusCode: 200;
+    @header eTag: string;
+    @body pet: Pet;
+  } | {
+    @statusCode statusCode: 404;
+  };
+
+  @post
```
Comment (timotheeguerin) [inline docs/getting-started/getting-started-http-06-headers.md:36]: Example here has a lot of thisn that don't seem related to `@header`.

File: docs/getting-started/getting-started-http-06-headers.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+---
+id: getting-started-http-06-headers
+title: Headers
+---
+
+# Headers
+
+Headers are used to pass additional information with the request or response. Model properties and parameters that should be passed in a header use the `@header` decorator. The decorator takes the header name as a parameter. If a header name is not provided, it is inferred from the property or parameter name.
+
+## What is `etag`?
```
Comment (timotheeguerin) [inline docs/getting-started/getting-started-http-06-headers.md:10]: Is it necessary to go over what an eTag header here is? If the goal is to just add example on how to specify eTag mnaybe could have an appendix at the end to explain what it is instead of starting with it.

File: docs/getting-started/getting-started-http-07-request-response-bodies.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+title: Request and Response Bodies
+---
+
+# Request and Response Bodies
+
+Request and response bodies can be declared explicitly using the `@body` decorator. This decorator helps to clearly indicate which part of the model is the body of the request or response. While it may not change the API's functionality, it provides several benefits:
+
+1. **Clarity and readability**: Using the `@body` decorator makes it explicit which part of the model is intended to be the body. This can improve the readability of the code, making it easier for developers to understand the structure of the API.
+2. **Consistency**: Applying the `@body` decorator consistently across your API definitions can help maintain a uniform style. This can be particularly useful in larger projects with multiple contributors.
+3. **Tooling and documentation**: Some tools and documentation generators may rely on the `@body` decorator to produce more accurate and detailed outputs. By explicitly marking the body, you ensure that these tools can correctly interpret and document your API.
```
Comment (timotheeguerin) [inline docs/getting-started/getting-started-http-07-request-response-bodies.md:12]: Not sure those points are really accurate.

Body allows you to specify an explicit property that represent exactly the body. But other uses are just as valid

Comment (timotheeguerin) [reply docs/getting-started/getting-started-http-07-request-response-bodies.md:12]: should also mention `@bodyRoot`

File: docs/getting-started/getting-started-http-10-summary.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
...NwYWPlAKHGcuUAiWVudW0gyCjGEyAgdjE6ICIxLjAuMOQAj8USMjogIjLKEn3kAQQgIEByb3V0ZSgiL3BldHPEfSAgzWnJUkBhZGTrAJEudjHFLSAgbW9kZWzEK8gqICBAbWluTGVuZ3RoKMgmxlc6IHN0cugBgcosVmFsdWUoMMkrQG1heMYUMTDKFmFnZTogaW50MzLLP2tpbmQ6ICJkb2ciIHwgImNhdMUIZmlzaMUJYmlyZMUJcmVwdGlsZeQCRsQ45wD%2FICBvcCBsaXN0KEBxdWVyeSBza2lwP8dnLCDHFXRvyRQpOusA53N0YXR1c0NvZGUgygs6IDIwMMdmICBAYm9keSDkAV065AEkW13HGn3pAMRvcCByZWFkKEBwYXRoxCpJZMdsLCBAaGVhZGVyIGlmTWF0Y2g%2F6AFA%2FwCG8gCGx0hlVGFnyETyAKPlAKLHF30gfN9jZTogNDA0z0ZlcnJvcjogTm90Rm91bmRFxA%2FxAPJAcG9zdMYLb3AgY3JlYXRlKO4Ahv8A4cV%2BMjAx%2FwCtzS808QGW5wCtVmFsaWRhdGlvbu0Ar99UxVQ12FRJbnRlcm5hbOQEFGVy%2BAEHdeoBBnVwZOUBBvQCBe8BGshx6wCg%2FwIIx3vsAKjGcGRQ7gH2xAF932D8AgXxAQj0AgnfZddl9ADF%2FQHB32fVZ%2BwB0NFn%2BgHUxWvqAdhkZWxldGXGDW9wIMYP8wHb%2FwLl5wG%2B6AFe%2FwKR%2FwOS9QOS31L%2FAuP9AuPpBWLmBh9Db21tb25QYXJhbWV0ZXLqBlbkARXHC3NlZ21lbnQoIuoGiyDtATDqAWLwBowyxy7VT3RveclPdG95zE%2FqBwfWUuYAvVRveegAsO8GxcpA5QEVyjPvApPIP2PlAVMiVkFMSURBVElPTl9FUlJPUugGlW1lc3PlBubLYSAgZGV0YWlsc8gW5wZD13ntAejPd05PVF9GT1VORNxw11jzAe7PXklOVEVSTkFMX1NFUlZFUv8A2stqYXV0b1LkCIbEDmnEaGbkCH1Ub3lPcGVy5QFF%2BgiG6QH6Z2XnBU1nZXRUb3koLi4u8AJ1KTrEVSB88QMN31zqBanmBTbXXyzkAl7FY9lp5QD1fQ0K&e=%40typespec%2Fopenapi3&options=%7B%7D).
```
Comment (timotheeguerin) [inline docs/getting-started/getting-started-http-10-summary.md:167]: I would recommend using this instead of encoded playground whihc might be hard to maintain/review

````
```tsp tryit="{"emit": ["@typespec/openapi3"]}"

````

Existing example hhere https://typespec.io/multi-protocol

The same could be done for all the complete examples in the guide too

File: docs/getting-started/getting-started-http-04-resources-routes.md (hunk 2 above)
Comment (mario-guerra) [reply docs/getting-started/getting-started-http-04-resources-routes.md]: Change made in updated commit.

File: docs/getting-started/getting-started-http-03-versioning.md (hunk 1 above)
Comment (mario-guerra) [reply docs/getting-started/getting-started-http-03-versioning.md:1]: I restructured the entire thing, I want to encourage users to think about versioning up front rather than an afterthought, it will save them pain in the long run.

File: docs/getting-started/getting-started-http-04-resources-routes.md (hunk 2 above)
Comment (mario-guerra) [reply docs/getting-started/getting-started-http-04-resources-routes.md]: done

File: docs/getting-started/getting-started-http-06-headers.md (hunk 3 above)
Comment (mario-guerra) [reply docs/getting-started/getting-started-http-06-headers.md]: I agree, I revised this section to include info on auth and content type headers.

File: docs/getting-started/getting-started-http-06-headers.md (hunk 4 above)
Comment (mario-guerra) [reply docs/getting-started/getting-started-http-06-headers.md]: I think etag has a place in the expanded context, let me know what you think.

File: docs/getting-started/getting-started-http-07-request-response-bodies.md (hunk 5 above)
Comment (mario-guerra) [reply docs/getting-started/getting-started-http-07-request-response-bodies.md]: Added info on @bodyRoot, although I left the other points intact. I'm open to revising if you think there are other points that are more relevant.

File: docs/getting-started/getting-started-http-10-summary.md (hunk 6 above)
Comment (mario-guerra) [reply docs/getting-started/getting-started-http-10-summary.md]: done

File: packages/website/sidebars.ts [hunk 7]
```diff
@@ -55,6 +55,17 @@ const sidebars: SidebarsConfig = {
       items: [
         "getting-started/getting-started",
         "getting-started/getting-started-http",
+        "getting-started/getting-started-http-01-setup",
```
Comment (timotheeguerin) [inline packages/website/sidebars.ts:58]: do you think its maybe better to move this into a sub folder to leave space for other guides?

Comment (mario-guerra) [reply packages/website/sidebars.ts:58]: That's fine with me, how does that impact navigation on the website?

Comment (timotheeguerin) [reply packages/website/sidebars.ts:58]: would show up as a folder on the website as well, you then use 
```ts
{
              "type": "autogenerated",
              "dirName": "getting-started/getting-started-http"
            }
```
to let docusaurus create that list automatically for this folder

Comment (mario-guerra) [reply packages/website/sidebars.ts]: Okay I made this change based on the similar structure done for 'standard-library', hopefully I did it right.

**Comment by timotheeguerin** [comment]
looks like the website build failed, some markdown links are broken

**Comment by mario-guerra** [comment]
> looks like the website build failed, some markdown links are broken

Where can I see what's broken?

**Comment by timotheeguerin** [comment]
in the CI logs https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3958634&view=logs&jobId=096134df-e707-5fe2-e74b-b0d2ee3c08d1

or running `npm run build` locally in the packages/website folder

**Comment by mario-guerra** [comment]
@timotheeguerin  it doesn't look like the nesting worked, did I not set it up correctly?

image

**Comment by mario-guerra** [comment]
@timotheeguerin I've tried a few different things, I'm not able to get the guide to nest in a sub-folder on the website. Can you take a look at the PR and let me know what I'm missing?

**Review by timotheeguerin — APPROVED** [review]
